aws_lambda_builders/workflows/nodejs_npm/utils.py

```diff
@@ -0,0 +1,40 @@
+"""
+Commonly used utilities
+"""
+
+import os
+import tarfile
+import subprocess
+import sys
+
+
+class OSUtils(object):
+
+    """
+    Wrapper around file system functions, to make it easy to
+    unit test actions in memory
+    """
+
+    def extract_tarfile(self, tarfile_path, unpack_dir):
+        with tarfile.open(tarfile_path, 'r:*') as tar:
+            tar.extractall(unpack_dir)
+
+    def joinpath(self, *args):
+        return os.path.join(*args)
+
+    def popen(self, command, stdout=None, stderr=None, env=None, cwd=None):
+        p = subprocess.Popen(command, stdout=stdout, stderr=stderr, env=env, cwd=cwd)
+        return p
+
+    @property
+    def pipe(self):
+        return subprocess.PIPE
+
+    def dirname(self, path):
+        return os.path.dirname(path)
+
+    def abspath(self, path):
+        return os.path.abspath(path)
+
+    def is_windows(self):
+        return sys.platform.lower().startswith('win')
```

Comment:
What about cygwin?
https://docs.python.org/3/library/sys.html#sys.platform

Can we follow was SAM CLI does for this already: https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/lib/samlib/cloudformation_command.py#L33

Reply:
I've changed the OS check to be the same as in the SAM CLI here https://github.com/gojko/aws-lambda-builders/commit/2c5d7807a95f38729ce0dc1dc4f4bf776b3b7a9c. Checking for .exe doesn't matter here as we know it's a .cmd, but I guess this can be handled in a more generic way as part of the path resolver PR (https://github.com/awslabs/aws-lambda-builders/pull/35)